### ISSUE TITLE
FSE: Cover Block does not render children blocks in the page editor, and the style is incorrect.

### ISSUE THREAD

**Issue by sarahfg** [Automattic/wp-calypso#35429]
#### Steps to reproduce
1. Start a FSE site as instructed by horizon
2. Add a full width Cover Block to the header
3. Place Site Title, Description, Menu over Cover Image within the Cover Image Block
4. Save changes and head back to a page editor

#### What I expected

I expected to see my header as I've set it in the header editor previewed on my page and as it's shown on the site

#### What happened instead

In the header editor the menu looks like it's text-align left.

In the page editor it though I have added only the cover image to the header. No block that is within the cover block will show.

On the site itself it looks as I'd expect.

#### Browser / OS version

Chrome/ Mac

#### Screenshot / Video

In the header editor
![](https://cld.wthms.co/nzoMQI+)

(Image link: https://cld.wthms.co/nzoMQI )

In the page editor

![](https://cld.wthms.co/h2fsZZ+)

(Image link: https://cld.wthms.co/h2fsZZ )

On the site itself

![](https://cld.wthms.co/3z7Gk6+)

(Image link: https://cld.wthms.co/3z7Gk6 )

**Comment by sarahfg** [comment]
A similar issue appears when adding a cover image to the footer

Footer Editor

![](https://cld.wthms.co/c7xFjk+)
(Image link: https://cld.wthms.co/c7xFjk )

Page Editor

![](https://cld.wthms.co/NFAXzY+)
(Image link: https://cld.wthms.co/NFAXzY )

Actual Site

![](https://cld.wthms.co/6qZ44B+)
(Image link: https://cld.wthms.co/6qZ44B )

**Comment by noahtallen** [comment]
I was able to reproduce this:
- None of the nested content displays in the page editor
- The block does not take up the full width in the page editor
- The alignment of nested blocks in the template editor does not reflect the front end (i.e. the nav menu is left aligned in the template editor, but center aligned on the front end)

**Comment by noahtallen** [comment]
After some more debugging, I fixed the second issue.

Per the first issue, it's super strange. It looks like if you add a paragraph block as the first block, it will show up in the page editor. But, if you add one of our dynamic blocks like navigation, title, or description, those blocks do not. To make it more confusing, if you add any of our dynamic blocks _before_ the paragraph block, the paragraph block stops showing up.

**Comment by noahtallen** [comment]
The _reason_ this happens is that [this action](https://github.com/WordPress/gutenberg/blob/5920ab8620e0a5ea6ed810e399dc42fd94d99abd/docs/designers-developers/developers/data/data-core-block-editor.md#replaceInnerBlocks) gets called, replacing all of the blocks inside the cover block. Not sure what triggers it, though!

**Comment by noahtallen** [comment]
In gutenberg, the bug is happening [here](https://github.com/WordPress/gutenberg/blob/15f5ff44609df48a47dee1d3e750470cdda50a5e/packages/block-editor/src/components/inner-blocks/index.js#L79-L88). The Cover block uses `InnerBlocks` to render its children. I verified in the debugger that it does a check here, and for some reason, the template does not have all of the blocks in it as it should. It then replaces the Cover's blocks such that we see the bug above. 

```
	synchronizeBlocksWithTemplate() {
		const { template, block, replaceInnerBlocks } = this.props;
		const { innerBlocks } = block;


		// Synchronize with templates. If the next set differs, replace.
		const nextBlocks = synchronizeBlocksWithTemplate( innerBlocks, template );
		if ( ! isEqual( nextBlocks, innerBlocks	) ) {
			replaceInnerBlocks( nextBlocks );
		}
	}
```

**Comment by noahtallen** [comment]
I was able to trace this back to a PR in gutenberg which fixes the issue: https://github.com/WordPress/gutenberg/pull/16973. I pulled those changes to my test environment, and this bug is fixed!

**Comment by Copons** [comment]
The Cover block happens to have an odd behaviour on Varia, and not just with FSE enabled (cc @iamtakashi):

This is what it looks like in the Header template. Note how the overlay covers the following block.
![Screenshot 2019-08-28 at 13 57 11](https://user-images.githubusercontent.com/2070010/63857851-533b7f00-c99c-11e9-8209-5790b90e9ccc.png)

This is how it looks like in the Page editor. (The Cover was intentionally made full width)
It goes off the edges, and completely covers the following block.
![Screenshot 2019-08-28 at 13 57 46](https://user-images.githubusercontent.com/2070010/63857989-9a297480-c99c-11e9-9d47-55b1c0c0df0a.png)

This is how it looks like in the front-end. LGTM!
![Screenshot 2019-08-28 at 13 57 18](https://user-images.githubusercontent.com/2070010/63858013-a8779080-c99c-11e9-8c24-6e3a6ccb63e1.png)

For reference, this is how it looks like in the editor with Twenty Nineteen.
![Screenshot 2019-08-28 at 14 00 35](https://user-images.githubusercontent.com/2070010/63858048-b88f7000-c99c-11e9-9903-e552a3eca953.png)

**Comment by iamtakashi** [comment]
> The Cover block happens to have an odd behaviour on Varia, and not just with FSE enabled

@Copons Sorry, I'm struggling to see the issue you're pointing 😅I'll ping you tomorrow!

**Comment by Copons** [comment]
@iamtakashi yes those screens were totally unclear.

Basically, the Cover block overlaps the neighbouring blocks.
Take a look at these screens, that have visible borders (Cover block selected/unselected):

![Screenshot 2019-08-29 at 10 57 46](https://user-images.githubusercontent.com/2070010/63930894-18911f80-ca4c-11e9-9a73-7e2e45821dd1.png)

![Screenshot 2019-08-29 at 10 57 56](https://user-images.githubusercontent.com/2070010/63930897-1c24a680-ca4c-11e9-9e1f-5eb955e08ff8.png)

Also, for some reasons, the Cover block coloured overlay is semi-transparent when the block is NOT full width, but turns into fully opaque when the block is full width.
(Not shown in these screens)

**Comment by iamtakashi** [comment]
@Copons Thanks, I see it now. This is something we need to fix in Varia as you expected. Cover block introduced the height setting with this update, but themes are currently controlling that in CSS, that's why the overlay bleeding out the cover block. We'll fix that.

**Comment by iamtakashi** [comment]
@Copons The cover block thing is no longer issue with 6.5 🎉

**Comment by Copons** [comment]
@iamtakashi Yes! The cover issue is fixed by 6.5!

This issue, though, is still happening, as the cover block stylesheet overrides a bunch of rules (e.g. `h1`, `a`, etc) that affect our FSE blocks.
I'm working on a quick fix right now!
